### PR TITLE
feat: smoke tests

### PR DESCRIPTION
**Pull request by 0marperez**
## Description of changes

Reintroduces support for [smoke tests](https://github.com/awslabs/aws-sdk-kotlin/pull/1388) which was reverted in #1421. With the addition of smoke test e2e tests running with `aws.kotlin.native` turned off.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        assertContains(smokeTestRunnerOutput, "ok SuccessService SuccessTest - no error expected from service # skip")
+        assertContains(smokeTestRunnerOutput, "ok SuccessService SuccessTestWithTags - no error expected from service # skip")
+    }
+}
+
+private fun runSmokeTests(service: String, envVars: Map<String, String> = emptyMap()): String {
+    val sdkRootDir = System.getProperty("user.dir") + "/../../../"
+    val runner = GradleRunner.create()
+        .withProjectDir(File(sdkRootDir))
+        .withArguments(listOf("-Paws.kotlin.native=false", ":tests:codegen:smoke-tests:services:$service:smokeTest"))
```
Comment (lauzadis) [inline tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt:46]: Nice find!

**Review by lauzadis — APPROVED** [review]

File: tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt (hunk 1 above)
Comment (0marperez) [reply tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt:46]: Thanks!

File: tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt [hunk 2]
```diff
@@ -43,7 +43,7 @@ private fun runSmokeTests(service: String, envVars: Map<String, String> = emptyM
     val sdkRootDir = System.getProperty("user.dir") + "/../../../"
     val runner = GradleRunner.create()
         .withProjectDir(File(sdkRootDir))
-        .withArguments(":tests:codegen:smoke-tests:services:$service:smokeTest")
+        .withArguments(listOf("-Paws.kotlin.native=false", ":tests:codegen:smoke-tests:services:$service:smokeTest"))
```
Comment (ianbotsf) [inline tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt:46]: **Nit**: A `vararg` overload of `withArguments` exists so `listOf` could be removed.

```kotlin
.withArguments("-Paws.kotlin.native=false", ":tests:codegen:smoke-tests:services:$service:smokeTest")
```

Comment (ianbotsf) [inline tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt:46]: **Question**: Do we have a task or something that tracks removing all the places we've disabled native builds? If not, I suggest we add a `FIXME` or something else here to increase the likelihood we remember to remove this once Kotlin/Native is ready.

**Review by ianbotsf — APPROVED** [review]

File: tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt (hunk 2 above)
Comment (0marperez) [reply tests/codegen/smoke-tests/src/test/kotlin/SmokeTestE2ETest.kt:46]: That's a good point. I added a subtask to our Kotlin Native task so we remember.

**Pull request merged**